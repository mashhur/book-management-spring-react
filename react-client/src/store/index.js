import { configureStore } from "@reduxjs/toolkit";
import locationReducer from "./slices/locationSlice";
import userReducer from "./slices/userSlice";

export const store = configureStore({
	reducer: {
		location: locationReducer,
		user: userReducer,
	},
});
