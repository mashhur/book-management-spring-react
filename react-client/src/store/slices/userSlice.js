import { createSlice } from "@reduxjs/toolkit";

const inituser = JSON.parse(localStorage.getItem("user"));
const user = inituser || {
	cid: "",
	fullName: "",
	phoneNumber: "",
};
const initialState = {
	auth: localStorage.getItem("auth") || false,
	user: user,
	address: {
		aid: "",
		customerName: "",
		phoneNumber: "",
		address: "",
		mstate: "",
		pincode: "",
		customerId: "",
	},
	books: [],
	cart: [],
};

export const userSlice = createSlice({
	name: "user",
	initialState,
	reducers: {
		changeAuth: (state) => {
			state.auth = true;
		},
		setUser: (state, action) => {
			state.user = action.payload;
		},
		setAddress(state, action) {
			state.address = action.payload;
		},
		addBook(state, action) {
			state.books = [];
			state.books.push(action.payload);
		},
		addBooktoCart(state, action) {
			state.cart.push(action.payload);
		},
		addtoBuy(state) {
			const cart = state.cart;
			state.books = cart;
			state.cart = [];
		},
		clearOrderDetails(state) {
			state.books = [];
			state.address = {};
		},
		removeBook(state, action) {
			state.cart.splice(action.payload, 1);
		},
		clearAuth(state) {
			state.auth = false;
			state.user = {
				cid: "",
				fullName: "",
				phoneNumber: "",
			};
		},
	},
});

export const {
	changeAuth,
	setUser,
	setAddress,
	addBook,
	clearOrderDetails,
	removeBook,
	addBooktoCart,
	addtoBuy,
	clearAuth,
} = userSlice.actions;

export default userSlice.reducer;
