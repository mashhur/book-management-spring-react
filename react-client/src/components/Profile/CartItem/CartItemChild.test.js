import React from "react";
import { mount, shallow } from "enzyme";
import {
	checkProp,
	findByTestAttr,
} from "../../../tests/testUtils";
import ItemChild from "./CartItemChild";
import { Provider } from "react-redux";
import { createTestStore } from "../../../store";
import { MdOutlineRemoveCircleOutline } from "react-icons/md";
import { addBooktoCart } from "../../../store/slices/userSlice";

const defaultProps = {
	book: {
		bid: 1,
		title: "",
		price: "",
		description: "",
		cover: "",
	},
};

const store = createTestStore();
const setup = (props = {}) => {
	const setUpProps = { ...defaultProps, ...props };
	return mount(
		<Provider store={store}>
			<ItemChild {...setUpProps} />
		</Provider>
	);
};

describe("Address item component test casses", () => {
	it("renders without any error", async () => {
		const wrapper = setup();
		const component = await findByTestAttr(
			wrapper,
			"component-cart-item"
		);
		expect(component.length).toBe(1);
	});

	it("on click remove item from store", () => {
		store.dispatch(addBooktoCart(defaultProps.book));
		const wrapper = setup();
		const component = wrapper.find(
			MdOutlineRemoveCircleOutline
		);
		component.simulate("click");
		expect(store.getState().user.cart.length).toBe(0);
	});

	it("doesnot throw warning with expected props", async () => {
		checkProp(ItemChild, defaultProps);
	});

	it("should match the deafult snapshot", () => {
		const wrapper = setup();
		expect(wrapper).toMatchSnapshot();
	});
});
