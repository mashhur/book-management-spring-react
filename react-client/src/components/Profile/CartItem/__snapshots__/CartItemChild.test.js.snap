// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Address item component test casses should match the deafult snapshot 1`] = `ReactWrapper {}`;
